src/app/qgisapp.cpp

```diff
@@ -10908,19 +10908,37 @@ void QgisApp::toggleMapTips( bool enabled )
 
 void QgisApp::toggleEditing()
 {
-  QgsMapLayer *currentLayer =  activeLayer();
-  if ( currentLayer && currentLayer->supportsEditing() )
+  const QList<QgsMapLayer *> layerList = layerTreeView()->selectedLayers();
+  if ( !layerList.isEmpty() )
   {
-    toggleEditing( currentLayer, true );
+    // if there are selected layers, try to toggle those
+    const bool shouldStartEditing = !mActionToggleEditing->isChecked();
+    for ( const auto layer : layerList )
+    {
+      if ( layer->supportsEditing() &&
+           shouldStartEditing == layer->isEditable() )
```

Comment:
Shouldn't this be
```suggestion
           shouldStartEditing != layer->isEditable() )
```
?

Reply:
I was really confused by the fact that your correction seems right while it is working as expected!
Then I realized that `shouldStartEditing` is actually `shouldStopEditing` because the action's checked status is read *after* it has been toggled by triggering it!
I should fix the wording for readability.